OrbitCore/ConnectionManager.cpp

```diff
@@ -226,6 +227,14 @@ void ConnectionManager::SetupClientCallbacks() {
     }
   });
 
+  GTcpClient->AddCallback(Msg_KeyAndString, [=](const Message& a_Msg) {
```

Comment:
To minimize (but not completely solve) threading issues, you could use GTcpClient->AddMainThreadCallback here.  This will defer the processing of the Msg_KeyAndString messages to the main thread.  We should probably make the string manager thread safe as well.

Reply:
I there still any benefit to using AddMainThreadCallback when string manager is thread safe? I'm assuming no, but please let me know if I should still address this.